pkg/southbound/gnmi/subscription.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"strings"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+
+	"github.com/onosproject/onos-config/pkg/utils"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// NewSubscription creates a new subscription
+func NewSubscription(options ...func(subscription *Subscription)) *Subscription {
+	subscription := &Subscription{}
+	for _, option := range options {
+		option(subscription)
+	}
+
+	return subscription
+}
+
+// Subscription subscription request information
+type Subscription struct {
+	updatesOnly       bool
+	prefix            string
+	mode              string
+	streamMode        string
+	sampleInterval    uint64
+	heartbeatInterval uint64
+	paths             [][]string
+	origin            string
+}
+
+// WithUpdatesOnly sets updatesOnly field
+func WithUpdatesOnly(updatesOnly bool) func(subscription *Subscription) {
```

Comment:
Since all these option functions just set fields in the struct, we can/should probably just remove them and define the `Subscription` struct explicitly.